static/js/formula.js

```diff
@@ -260,25 +258,58 @@ function is_valid_formula(str) {
 
         //If the total length of segments obtained from the string are not equal to the original
         //string, then there is an invalid piece.
-        isValid = (acceptedSegmentLengths == (str.length-form_qty.length));
+        if(/^\d+/g.test(str)){
+            isValid = (acceptedSegmentLengths == (str.length-form_qty.length));
+        }else{
+            isValid = acceptedSegmentLengths == str.length;
+        }
     }
 
     return isValid.valueOf();
 }
 
+/**
+ * Breaks a segment string down into an array containing the element and quantity
+ * index 0 = element + quantity
+ * index 1 = element
+ * index 2 = quantity
+ *
+ * @param segment string to break down
+ * @returns {Array|{index: number, input: string}} array whos contents are:
+ *      [0] = element+quantity
+ *      [1] = element
+ *      [2] = quantity
+ *
+ */
+function segment_to_pieces(segment){
+    return segment.match(/([A-Z]{1}[a-z]?)(\d*)/);
+}
+
+/**
+ * Breaks a formula string down into an array of elements and their respective quantities
+ * @param str formula string to break down.
+ * @returns {Array|{index: number, input: string}} an array of elements concatenated with their quantities
+ */
+function string_to_compound_segments(str){
+    return str.match(/[A-Z]{1}[a-z]?\d*/g);
+}
+
+
+/**
+ * Obtains the occurrences of a given formula.
+ * @param str Formula string
+ * @returns {number} number of times formula occurs.
+ */
 function front_number(str){
-    var form_qty = 1;
 
+    var form_qty = 1;
     if(/^\d+/g.test(str)){
         //check if input had a qty at the beginning
-        form_qty = str.match(/^\d+/)[0].valueOf();
-        //window.alert(form_qty + " length = " + form_qty.length);
+        form_qty = parseInt(str.match(/^\d+/)[0]);
     }
-
     return form_qty;
 }
```

Comment:
Gets the quantity of a formula off of the front of a string. If a formula does not have a leading number, its quantity is 1. The function will return 1 by default to reflect the default "1" quantity for formulae.